### ISSUE TITLE
Multimodal Few-Shot Learning with Frozen Language Models

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#147]
## TL; DR

- Train a vision encoder compatible to the pretrained auto-regressive language model (sharing the same latent space) such that the language model can generate appropriate captions for the input image.
- Explore its capacity to quickly adapt to diverse novel tasks (VQA, image classification) in few-shot manner.

## Motivation

- Auto-regressive language models are known to be excellent few-shot learners, but their modality was mainly confined to language domain.
- Leverage well-trained language model for general visual understanding, in a simple and modulo (partial plug-in) fashion.

## Prior Work

- Language Models are Few-shot Learners (GPT-3)
- Prefix-tuning: optimize the prefix embedding (constant bias) for task-specific prior
- Multi-modal transformers

### Training

![image](https://user-images.githubusercontent.com/46308407/148062582-b5aef698-9edf-48e8-85c0-b9ad93ce487f.png)

- Grab an off-the-shelf auto-regressive language model (in this case, 7-billion param model trained on C4)
- Add a NF-ResNet-50 convolutional ***vision encoder*** that maps images to the latent space of the pretrained language model. (from scratch)
- Gradient flows through the language model (for log-likelihood objective in captioning) but the parameters are frozen. (hence ***Frozen***)
- Vision encoder is the only part that is updated during captioning pretraining.

![image](https://user-images.githubusercontent.com/46308407/148062909-5220bc0f-d2ba-435f-b039-84c80fe64966.png)

### Inference for Diverse Tasks with Varying Number of Shots

![image](https://user-images.githubusercontent.com/46308407/148062977-ae7b935f-3d59-4dc3-9f51-1c6eb5193bb6.png)

- Different interfaces for different tasks

![image](https://user-images.githubusercontent.com/46308407/148063042-82c1ea80-cdc7-48e8-bc9b-3798a86cded0.png)

- 3-shot inference.

- Note that during pretraining, *Conceptual Caption* dataset (3M image-caption pairs) is used, so only a single pair of image and text is provided.
- However, the model is fully modality-agnostic, so when multiple image-text pairs are provided interleaved (e.g., image-text-image-text-image-???) it can still produce reasonable output. This enables few-shot inference (without finetuning)

### Fast Adaptation

![image](https://user-images.githubusercontent.com/46308407/148063336-20003f64-dc7e-4d4e-8e51-6b328870c6f6.png)

- ***Frozen*** is trained on captioning task, but it can generalize to VQA with zero to few support samples (provided as sequence of tokens)
- scratch: language model from scratch
- finetuned: update language model during captioning
- train-blind: provide black-out images (all pixel values zeroed) during captioning to see if the model actually leverages visual information. (strong language models could simply output answers without referring to the image)
- VQA: include VQA dataset into captioning training stage.

### General Knowledge

- Conceptual Caption dataset (multi-modal) has all the unique names replaced with common proverbs (e.g., Elon Musk -> the person)
- Hence, when asked with knowledge-intensive questions (e.g., who invented the airplane? -> Wright brothers), the model has no way but to refer to the knowledge learned in ***language-only*** pretraining stage. (C4 has all the unique names preserved)

![image](https://user-images.githubusercontent.com/46308407/148063623-b3df13ed-7f84-4e80-b4e1-0e8b4e48034d.png)

- Note that ***Frozen*** outperforms its own baseline variants, though far from the SOTA.
- 400mLM: smaller language model (400M instead of default 7B)

### Fast Concept Binding

![image](https://user-images.githubusercontent.com/46308407/148064033-c240cf23-4684-4b55-8896-c3cf3ecd3c65.png)

- Replace the image class with non-sense words (new concept) and see how quickly the model picks up the new word.

![image](https://user-images.githubusercontent.com/46308407/148064069-366a0baa-7414-46a3-9448-471db660405e.png)

- Works somewhat well for 2-way few-shot setting, but not so well with 5-way setting. Providing 5 examples in a sequential manner (array of tokens) is currently impossible to effectively process with Tr.
- With *real-name*, the model can also refer to the semantics crystallized in the class label (e.g., blicket -> lion), hence better performance is expected.

![image](https://user-images.githubusercontent.com/46308407/148064242-07a54285-2f15-4977-a145-690668a43a2b.png)

- Fast-VQA tests the model's capacity to map the new concept quickly to the appropriate location in the language latent space, hence associate it with correct neighboring notions for answering questions regarding the concept.
- Fast-VQA-real provides real class labels, making it easier for the model to employ common knowledge encoded through language-only pretraining.

## Comments

- Provide simple, *working* baseline for open-ended multi-modal understanding.
- The performance is not satisfactory, but as the authors mention, this work has meanings in its *proof-of-concept*.
- Technical improvements could be made by devising effective finetuning schemes, as the current model simply *freezes* the language model and only train the vision encoder.
